amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/rm/rest/api/retentionschedule/RetentionScheduleActionRelation.java

```diff
@@ -0,0 +1,225 @@
+    {
+        checkNotBlank("retentionScheduleId", retentionScheduleId);
+        mandatory("parameters", parameters);
+        NodeRef retentionScheduleNodeRef = apiUtils.lookupAndValidateNodeType(retentionScheduleId, RecordsManagementModel.TYPE_DISPOSITION_SCHEDULE);
+        List<DispositionActionDefinition> actions = nodesModelFactory.getRetentionActions(retentionScheduleNodeRef);
+        List<RetentionScheduleActionDefinition> actionDefinitionList = actions.stream()
+                .map(nodesModelFactory::mapRetentionScheduleActionDefData)
+                .collect(Collectors.toList());
+        return CollectionWithPagingInfo.asPaged(parameters.getPaging(), actionDefinitionList, false,
+                actionDefinitionList.size());
+    }
+
+    /**
+     * this method is used to validate the order of the retention schedule step
+     * @param retentionScheduleNodeRef nodeRef
+     * @param retentionScheduleActionDefinition retention schedule action definition
+     */
+    private void retentionScheduleStepValidation(NodeRef retentionScheduleNodeRef, RetentionScheduleActionDefinition retentionScheduleActionDefinition)
+    {
+        List<DispositionActionDefinition> actions = nodesModelFactory.getRetentionActions(retentionScheduleNodeRef);
+        Set<String> completedActions = new HashSet<>();
+        if (!actions.isEmpty())
+        {
+            completedActions = actions.stream()
+                    .map(DispositionActionDefinition::getName)
+                    .collect(Collectors.toSet());
+        }
+
+        if (completedActions.contains(RetentionSteps.DESTROY.stepName))
+        {
+            throw new ConstraintViolatedException("Invalid Step - destroy action is already added . No other action is allowed after Destroy.");
+        }
+
+        if (checkStepAlreadyExists(completedActions, retentionScheduleActionDefinition))
+        {
+            throw new ConstraintViolatedException("Invalid Step - This step already exists. You can’t create it again. Only transfer action is allowed multiple times.");
+        }
+
+        if (firstStepValidation(actions, retentionScheduleActionDefinition))
+        {
+            throw new UnprocessableContentException("Invalid Step - cutoff or retain should be the first step");
+        }
+
+        if (isCutOffStepAllowed(completedActions, retentionScheduleActionDefinition))
+        {
+            throw new ConstraintViolatedException("Invalid Step - Can't use cutoff after transfer or accession");
+        }
+    }
+
+    /**
+     * this method is used to validate the request of the retention schedule
+     * @param retentionScheduleActionDefinition retention schedule action definition
+     */
+    private void retentionScheduleRequestValidation(RetentionScheduleActionDefinition retentionScheduleActionDefinition)
+    {
+        // step name validation
+        if (invalidStepNameCheck(retentionScheduleActionDefinition))
```

Comment:
Instead of passing the whole object of retentionScheduleActionDefinition as parameter in a function we can pass only the property that we want to check